internal/pkg/build/oci/oci.go

```diff
@@ -33,6 +33,16 @@ type ImageReference struct {
 	types.ImageReference
 }
 
+var archMap = map[string]string{
+	"amd64":   "amd64",
+	"armv6":   "arm",
+	"armv7":   "arm",
```

Comment:
Why is it OK for those two to share the same cache?

Reply:
https://go.dev/doc/install/source#environment
https://docs.docker.com/build/building/multi-platform/
we are acutally using goarch value here, but to map it to docker image archs, we need a map.

Reply:
To continue on this discussion, we should probably add the arch variants into consideration when computing the new digest, between armv6/armv7 here seems not correct and armv5 is missing, looks like the correct values are:
- `arm32v5`
- `arm32v6`
- `arm32v7`

Mapped to Go runtime variables:
- `arm32v5` :
  - `GOARCH = arm`
  - `GOARM = 5`
- `arm32v6` :
  - `GOARCH = arm`
  - `GOARM = 6`
- `arm32v7` :
  - `GOARCH = arm`
  - `GOARM = 7`

`types.SystemContext` has `VariantChoice` and for arm cases we would end up with:
- `arm32v5`:
  - `ArchitectureChoice = arm`
  - `VariantChoice = v5`
- `arm32v6`
  - `ArchitectureChoice = arm`
  - `VariantChoice = v6`
- `arm32v7`
  - `ArchitectureChoice = arm`
  - `VariantChoice = v7`
- `arm64v8` 
  - `ArchitectureChoice = arm64`
  - `VariantChoice = v8`